### ISSUE TITLE
Support for encodings defined in HTML meta tag

### ISSUE THREAD

**Issue by radiospiel** [sparklemotion/nokogiri#860]
Apparently nokogiri does not or not always supports encodings as defined via meta[http-equiv=content-type] tags. 

I build a workaround https://gist.github.com/radiospiel/5148046, where `Nokogiri::HTML.with_meta_encoding(data)` reparses the document if there is a meta encoding which does not match N:H:D.encoding.

It would be nice if `Nokogiri.HTML` would work like this - I think that this is correct behaviour as per some HTML specs, but I don't know if 

a) this is fitting in the nokogiri mind frame - it would change `Nokogiri.HTML` after all, and 
b) is only tested on my local machine as of now (MRI 1.9.2, nokogiri 1.5.6, libxml 2.7.3(?))

Does anyone want to take over? Can I somehow help?

**Comment by flavorjones** [comment]
Hi,

Thanks for posting this suggestion. I'm very sorry for not responding for so long.

Can you provide some test cases demonstrating how the proposed feature works? I'd like to understand the problem you're trying to solve, and then determine how best to solve it.

**Comment by radiospiel** [comment]
Hey @flavorjones no problem at all. Just found about your answer after only 10 months. In any case I consider this to not be an urgent issue, since most people are serving UTF-8 these days anyways. However, assume the following scenario:

- The web server serves a static file. Due to the webserver configuration the Content-Type header is set to chartype "utf-8". However, the person that created that specific file knows better and sets the documents content type to ISO-8859-1 `<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1">`

I would suggest nokogiri to handle this case (and reencode the source string as ISO-8859-1 before handing it off to the HTML parsing code, otherwise non-ascii characters, like "Ä", would fail. I encountered this a couple of times when I tried to scrape some websites with horrible, probably word-based layout. But as I said before: I don't think this is urgent in any way - and also I don't know whether or not nokogiri changed its behaviour in the meantime.
